joblib/numpy_pickle.py

```diff
@@ -270,6 +273,28 @@ def save(self, obj):
                         traceback.format_exc()))
         return Pickler.save(self, obj)
 
+    def save_bytes(self, obj):
+        """Strongly inspired from python 2.7 pickle.Pickle.save_bytes"""
+        if self.bin:
+            n = len(obj)
+            if n < 256:
+                self.write(pickle.SHORT_BINSTRING + asbytes(chr(n)) + obj)
+            else:
+                self.write(pickle.BINSTRING + struct.pack("<i", n) + obj)
```

Comment:
It would be good to benchmark the 2 approaches with `memory_profiler` on the following script:

```python
import joblib
import numpy as np

a = np.zeros(int(1e9), dtype=np.uint8)  # 1GB
joblib.dump(a, '/tmp/a.pkl')
```

I am pretty sure that the Python 2.7 version is forcing an unnecessary memory copy.